### PR TITLE
Fix measure-complexity command

### LINKED ISSUE #5134: Judging by the current code --iterations option is ignored
https://github.com/dafny-lang/dafny/blob/master/Source/DafnyDriver/Commands/MeasureComplexityCommand.cs#L42

We should make a Boogie change to update VerificationTask so it's cloneable. Dafny can then clone VerificationTasks when it sees --iterations is used, and run each clone. There's a request to output for each iteration which --randomSeed argument should be used to reproduce that iterations, so we can do that in the meantime: https://github.com/dafny-lang/dafny/issues/5126#issuecomment-1970264367

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Description
- Fix an unreleased bug that could cause `dafny verify` to stall
- Give meaning to `--iterations` and `--random-seed` options for `measure-complexity`
- Do iterations one at a time
- Report "Starting verification of iteration {iteration} with seed {iterationSeed}"); at the start of each iteration
- Add some references to the used random seed in verification logging

### How has this been tested?
- Updated `VerificationLogger.dfy` test

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyDriver/CliCompilation.cs [hunk 1]
```diff
@@ -236,17 +226,17 @@ public class CliCompilation : IDisposable {
       }
       yield return new CanVerifyResult(canVerify, results.CompletedParts.Select(c => new VerificationTaskResult(c.Task, c.Result.Result)).ToList());
 
-      canVerifyResults.Remove(canVerify); // Free memory
+      canVerifyResults.Remove(canVerify); //].CompletedParts.Clear(); // Free memory
```
Comment (atomb) [inline Source/DafnyDriver/CliCompilation.cs:229]: Can this be removed now?

**Review by atomb — CHANGES_REQUESTED** [review]
This is great. I just saw one tiny case of an obsolete comment.

**Comment by atomb** [comment]
> This is great. I just saw one tiny case of an obsolete comment.

And fixing the failing tests, of course. None of them look particularly serious, though.

File: Source/DafnyDriver/CliCompilation.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyDriver/CliCompilation.cs]: Done

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/logger/CSVLogger.dfy [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: %exits-with 4 %baredafny verify --log-format:csv";"LogFileName="%t.csv" "%s"
+// RUN: %exits-with 4 %baredafny verify --log-format:csv";"LogFileName="%t.csv" "%s" > %t
```
Comment (atomb) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/logger/CSVLogger.dfy:1]: If the file `%t` ever used for anything?

**Review by atomb — APPROVED** [review]
I had one tiny, non-blocking question. Otherwise it looks great.

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/logger/CSVLogger.dfy (hunk 2 above)
Comment (keyboardDrummer) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/logger/CSVLogger.dfy:1]: I guess not :-D

**Pull request merged**